hibernate-core/src/main/java/org/hibernate/internal/AbstractSharedSessionContract.java

```diff
@@ -248,6 +248,14 @@ protected final void setUpMultitenancy(SessionFactoryImplementor factory, LoadQu
 		}
 	}
 
+	protected final void setupAutoEnabledFilters(SessionFactoryImplementor factory, LoadQueryInfluencers loadQueryInfluencers) {
+		factory.getDefinedFilterNames()
```

Comment:
Since this is executed whenever a session is created, I would prefer to have `SessionFactoryImplementor` expose a `getAutoEnabledFilters()` that initializes the `enabledFilters` in the constructor of `LoadQueryInfluencers` instead.

Reply:
How about a `getAutoEnabledFilterNames()`? This way, the constructor of `LoadQueryInfluencers` could enable the filters looping on the names

Reply:
I would prefer avoiding to do hash map lookups if possible, so I'd rather not expose the names but filter definitions directly.